widget/richtext.go

```diff
@@ -562,7 +562,20 @@ func (r *textRenderer) layoutRow(texts []fyne.CanvasObject, align fyne.TextAlign
 				realign = true
 			}
 			size = s
-		} else {
+		} else if c, ok := text.(*fyne.Container); ok {
+			wid := c.Objects[0]
+			if link, ok := wid.(*Hyperlink); ok {
+				s, base := fyne.CurrentApp().Driver().RenderedTextSize(link.Text, theme.TextSize(), link.TextStyle)
+				if tallestBaseline == 0 {
+					tallestBaseline = base
+				} else if base != tallestBaseline {
+					tallestBaseline = fyne.Max(tallestBaseline, base)
+					realign = true
+				}
+				size = s
+			}
+		}
+		if size.IsZero() {
 			size = text.MinSize()
 		}
```

Comment:
That is quite a broad question. If they have a custom text segment using the style based API we created then it will align. If they implement a whole new thing returning a hierarchy of objects that somewhere within holds some text, no it may not align correctly.
One step at a time ;)

Reply:
Got it. Maybe it could be easier if `Visual()` method in `RichTextSegment` interface returns a custom interface like `RichTextVisual`:
```go
type RichTextVisual interface {
    fyne.CanvasObject
    TextMetrics() (size Size, baseline float32)
}
```
So developers could create custom segments. This addition would facilitate a lot the internal processes too as now we would not need to assert the type to find the data needed to get the text metrics. But, maybe that could be a later improvement.

Reply:
This assumes that the `TextSegment` is not customisable enough. At this time it is too early to say that it is not IMHO.

Reply:
Sounds good :)